index.bs

```diff
@@ -5638,6 +5639,107 @@ At this time, one [=credential property=] is defined: the [=resident key credent
+
+        :   <dfn>prfEval</dfn>
+        ::  One or two strings that are inputs on which to evaluate the freshly created PRF. Not all [=authenticators=] support evaluating the PRFs during credential creation so outputs may, or may not, be provided. If not, then an [=assertion=] is needed in order to obtain the outputs.
+    </div>
+
+: Client extension processing ([=registration extension|registration=])
+::
+       1. If {{AuthenticationExtensionsClientInputs/prfEval}} is present and {{AuthenticationExtensionsClientInputs/prfEnable}} is absent, or has the value [FALSE], return a {{DOMException}} whose name is “{{NotSupportedError}}”
+       1. If {{AuthenticationExtensionsClientInputs/prfEnable}} is present and has the value [TRUE]:
+           1. If {{AuthenticatorSelectionCriteria/userVerification}} has the value {{UserVerificationRequirement/preferred}}:
+               1. Return a {{DOMException}} whose name is “{{NotSupportedError}}”
+           1. Set `hmac-secret` to [TRUE] in the authenticator extensions input.
+           1. If {{AuthenticationExtensionsClientInputs/prfEval}} is present and a future extension to [[FIDO-CTAP]] permits evaluation of the PRF at creation time, configure `hmac-secret` inputs accordingly:
+               * Let `salt1` be the value of <code>SHA-256(UTF8Encode("WebAuthn PRF") || 0x00 || UTF8Encode({{AuthenticationExtensionsPRFInputs/input1}}))</code>.
+               * If {{AuthenticationExtensionsPRFInputs/input2}} is present, let `salt2` be the value of <code>SHA-256(UTF8Encode("WebAuthn PRF" || 0x00 || UTF8Encode({{AuthenticationExtensionsPRFInputs/input2}}))</code>.
+           1. Set {{AuthenticationExtensionsClientOutputs/prfEnabled}} to the value of `hmac-secret` in the authenticator extensions output. If not present, set {{AuthenticationExtensionsClientOutputs/prfEnabled}} to [FALSE].
+           1. Set {{AuthenticationExtensionsClientOutputs/prfOutputs}} to the PRF outputs, if any.
+
+: Client extension output ([=registration extension|registration=])
+:: <xmp class="idl">
+    partial dictionary AuthenticationExtensionsClientOutputs {
+        boolean prfEnabled;
+        AuthenticationExtensionsPRFOutputs prfOutputs;
+    };
+    </xmp>
+
+    <div dfn-type="dict-member" dfn-for="AuthenticationExtensionsClientOutputs">
+        :   <dfn>prfEnabled</dfn>
+        ::  [TRUE] if, and only if, the two PRFs are available for use with the created credential.
+
+        :   <dfn>prfOutputs</dfn>
+        ::  The results of evaluating the PRF for the inputs given in {{AuthenticationExtensionsClientInputs/prfEval}}. Outputs may not be available during [=registration=]; see comments in {{AuthenticationExtensionsClientInputs/prfEval}}.
+    </div>
+
+: Client extension input ([=authentication extension|authentication=])
+::  <xmp class="idl">
+    partial dictionary AuthenticationExtensionsClientInputs {
+        record<USVString, AuthenticationExtensionsPRFInputs> prfEvalByCredential;
+    };
```

Comment:
This member also needs to be named just `prf`. I think it can remain a `record`, but maybe there shouldn't be both this and the registration extension input with the same name but different types? @selfissued what do you think?

Reply:
Have defined  a dictionary, AuthenticationExtensionsPRFInputs, to hold the inputs. The member of AuthenticationExtensionsClientInputs is now just called `prf` and is an instance of that dictionary.